hosted/aks/helper/helper_cluster.go

```diff
@@ -34,7 +34,7 @@ var (
 )
 
 // CreateAKSHostedCluster creates the AKS cluster on Rancher
-func CreateAKSHostedCluster(client *rancher.Client, displayName, cloudCredentialID, k8sVersion, location string) (*management.Cluster, error) {
+func CreateAKSHostedCluster(client *rancher.Client, displayName, cloudCredentialID, k8sVersion, location string, updateFunc func(clusterConfig *aks.ClusterConfig)) (*management.Cluster, error) {
```

Comment:
But Chandan requested in the other PR to have the new flag optional.

Reply:
Yes a suggestion, @thehejik WDYT ?

Reply:
Personally I'd keep it mandatory as sending `nil` doesn't hurt so much and it is better for visibility. But I'm not an expert here.

Reply:
I agree with @thehejik on both the points. It would make sense to use the optional pattern if we were planning to pass more functions, which does not seem likely.

Reply:
Yes, Ok. We don't have optional parameters yet in any Create functions.